tests/test_plotfileparticledata.py

```diff
@@ -0,0 +1,163 @@
+import random
+import shutil
+import sys
+from dataclasses import dataclass
+
+import numpy as np
+import pytest
+
+# seed RNG to make test reproducible -- comment out this line to generate new
+# random particle positions every time.
+random.seed(1)
+
+# Import and initialize pyAMReX
+import amrex.space3d as amr
+
+# pyAMReX's pytest config uses a fixture that initializes amrex => check for
+# pytest before initializng pyAMReX
+if "pytest" not in sys.modules:
+    # guard against double initialization
+    amr.initialized() or amr.initialize([])
+
+# Particle container constructor -- depending on if gpus are available use the
+# GPU-enabled version. This uses the FHDeX Particle Container for the test
+if amr.Config.have_gpu:
+    PCC = amr.ParticleContainer_16_4_0_0_managed
+else:
+    PCC = amr.ParticleContainer_16_4_0_0_default
+
+
+@dataclass
+class Particle:
+    """
+    Helper class to work with particle data
+    """
+
+    x: float
+    y: float
+    z: float
+
+    idx: int
+
+    def to_soa(self, aos_numpy):
+        """Fill amr particle SoA with x, y, z, idx data"""
+        aos_numpy["x"] = self.x
+        aos_numpy["y"] = self.y
+        aos_numpy["z"] = self.z
+        aos_numpy["idata_0"] = self.idx
+
+
+def generate_test_particles(n_part):
+    """
+    Returns a list of test particles scattered throught the domain
+    """
+    particles = list()
+    generator = lambda: 1 - 2 * random.random()
+
+    for i in range(n_part):
+        particles.append(Particle(x=generator(), y=generator(), z=generator(), idx=i))
+
+    return particles
+
+
+def particle_container(Rpart, std_geometry, distmap, boxarr, std_real_box):
+    """
+    Generate a fresh particle container, containing copies from Rpart
+    """
+    pc = PCC(std_geometry, distmap, boxarr)
+
+    iseed = 1
+    myt = amr.ParticleInitType_16_4_0_0()
+    pc.init_random(len(Rpart), iseed, myt, False, std_real_box)
+
+    particles_tile_ct = 0
+    # assign some values to runtime components
+    for lvl in range(pc.finest_level + 1):
+        for pti in pc.iterator(pc, level=lvl):
+            aos = pti.aos()
+            aos_numpy = aos.to_numpy(copy=False)
+            for i, p in enumerate(aos_numpy):
+                Rpart[i + particles_tile_ct].to_soa(p)
+            particles_tile_ct += len(aos_numpy)
```

Comment:
I think there is an issue in this block, returning before it keeps the pc `.OK()`.

Reply:
This fixes the `pc.OK()` check for me:
```suggestion
    particles_tile_ct = 0
    # assign some values to runtime components
    for lvl in range(pc.finest_level + 1):
        for pti in pc.iterator(pc, level=lvl):
            aos = pti.aos()
            aos_numpy = aos.to_numpy(copy=True)
            for i, p in enumerate(aos_numpy):
                Rpart[i + particles_tile_ct].to_soa(p)
            particles_tile_ct += len(aos_numpy)
```

Reply:
Ah, I think the issue might be that some particles are out of bounds after the assignment here for their current boxes?

I added a `pc.redistribute()` before returning to make it pass, but we need to check the particles are not just removed now:
```suggestion
    particles_tile_ct = 0
    # assign some values to runtime components
    for lvl in range(pc.finest_level + 1):
        for pti in pc.iterator(pc, level=lvl):
            aos = pti.aos()
            aos_numpy = aos.to_numpy(copy=False)
            for i, p in enumerate(aos_numpy):
                Rpart[i + particles_tile_ct].to_soa(p)
            particles_tile_ct += len(aos_numpy)

    pc.redistribute()
    return pc     
```